unity-client/Assets/Scripts/MainScripts/DCL/Controllers/BuilderMode/States/EditorMode/Voxels/VoxelController.cs

```diff
@@ -0,0 +1,301 @@
+using DCL.Controllers;
+using DCL.Models;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class VoxelController : MonoBehaviour
+{
+    [Header("References")]
+    public VoxelPrefab voxelPrefab;
+    public BuilderInputWrapper builderInputWrapper;
+    public BuildModeController buildModeController;
+    public BuildEditorMode buildEditorMode;
+    public OutlinerController outlinerController;
+    public FreeCameraMovement freeCameraMovement;
+    
+    public LayerMask groundLayer;
+
+    DecentralandEntityToEdit lastVoxelCreated;
+
+    GameObject editionGO;
+    bool mousePressed = false, isVoxelModelActivated = false,isCreatingMultipleVoxels = false; 
+    Vector3Int lastVoxelPositionPressed;
+    Vector3 lastMousePosition;
+    Dictionary<Vector3Int, VoxelPrefab> createdVoxels = new Dictionary<Vector3Int, VoxelPrefab>();
+    ParcelScene currentScene;
+    private void Start()
+    {
+        builderInputWrapper.OnMouseDown += MouseDown;
+        builderInputWrapper.OnMouseUp += MouseUp;
+ 
+    }
+
+    private void Update()
+    {
+        if (mousePressed && isVoxelModelActivated && isCreatingMultipleVoxels)
+        {
+
+            bool fillVoxels = false;
+            Vector3Int currentPosition = Vector3Int.zero;
+            VoxelEntityHit voxelHit = buildModeController.GetCloserUnselectedVoxelEntityOnPointer();
+            if (voxelHit != null && voxelHit.entityHitted.tag == "Voxel" && !voxelHit.entityHitted.IsSelected)
```

Comment:
`"Voxel"`  is appearing in multiple locations. We should move it to a const in the config file